zap/src/main/java/org/zaproxy/zap/session/ScriptBasedSessionManagementMethodType.java

```diff
@@ -0,0 +1,668 @@
+            }
+            this.dynamicFieldsPanel.validateFields();
+        }
+
+        @Override
+        public void saveMethod() {
+            this.method.script = (ScriptWrapper) this.scriptsComboBox.getSelectedItem();
+            // This method will also be called when switching panels to save a temporary state so
+            // the state of the session management method might not be valid
+            if (this.dynamicFieldsPanel != null)
+                this.method.paramValues = this.dynamicFieldsPanel.getFieldValues();
+            else this.method.paramValues = Collections.emptyMap();
+        }
+
+        private void loadScript(ScriptWrapper scriptW, boolean adaptOldValues) {
+            SessionScript script = getScriptInterface(scriptW);
+
+            if (script == null) {
+                log.warn(
+                        "The script "
+                                + scriptW.getName()
+                                + " does not properly implement the Session Script interface.");
+                warnAndResetPanel(
+                        Constant.messages.getString(
+                                "session.method.script.dialog.error.text.interface",
+                                scriptW.getName()));
+                return;
+            }
+
+            try {
+                String[] requiredParams = script.getRequiredParamsNames();
+                String[] optionalParams = script.getOptionalParamsNames();
+
+                // If there's an already loaded script, make sure we save its values and _try_
+                // to place them in the new panel
+                Map<String, String> oldValues = null;
+                if (adaptOldValues && dynamicFieldsPanel != null) {
+                    oldValues = dynamicFieldsPanel.getFieldValues();
+                    if (log.isDebugEnabled()) log.debug("Trying to adapt old values: " + oldValues);
+                }
+
+                this.dynamicFieldsPanel = new DynamicFieldsPanel(requiredParams, optionalParams);
+                this.loadedScript = scriptW;
+                if (adaptOldValues && oldValues != null)
+                    this.dynamicFieldsPanel.bindFieldValues(oldValues);
+
+                this.dynamicContentPanel.removeAll();
+                this.dynamicContentPanel.add(dynamicFieldsPanel, BorderLayout.CENTER);
+                this.dynamicContentPanel.revalidate();
+
+            } catch (Exception e) {
+                getScriptsExtension().handleScriptException(scriptW, e);
+                log.error("Error while calling session management script", e);
+                warnAndResetPanel(
+                        Constant.messages.getString(
+                                "session.method.script.dialog.error.text.loading",
+                                ExceptionUtils.getRootCauseMessage(e)));
+            }
+        }
+
+        private void warnAndResetPanel(String errorMessage) {
```

Comment:
The panel is not being properly reset, if you load a script with parameters, cancel the dialogue and then go back it still shows the previous parameters/fields.

Reply:
No longer able to reproduce this, resolving.